### ISSUE TITLE
Rule Proposal: no-computed-in-data

### ISSUE THREAD

**Issue by scyclops** [vuejs/eslint-plugin-vue#958]
**Please describe what the rule should do:**

Ensure computed properties are not used in the data() function because data() runs before computed properties are calculated.  So computed properties are undefined when used in data().

**What category should the rule belong to?**

- [ ] Enforces code style
- [X] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue

<script>
export default {
    data(){
        return {
            value: 'hello ' + this.world
        }
    },
    computed: {
        world() {
            return 'world'
        }
    }
}
</script>


```

**Comment by ota-meshi** [comment]
The rule has been implemented #1653.
